std/algorithm/mutation.d

```diff
@@ -2389,6 +2374,55 @@ void swap(T)(ref T lhs, ref T rhs) if (is(typeof(lhs.proxySwap(rhs))))
     lhs.proxySwap(rhs);
 }
 
+import std.traits: isIntegral;
+
+/**
+Swaps two elements in-place of a random access range `r`,
+specified by their indices `i1` and `i2`.
+
+Params:
+    r  = a $(XREF_PACK_NAMED range,primitives,isRandomAccess,random access range)
+        with swappable elements
+    i1 = first index
+    i2 = second index
+*/
+void swapAt(R, I)(R r, I i1, I i2)
```

Comment:
We need the same style like `moveAt` :-/... This was just a question) Please add `in` for generic integral type: this will reduce template bloat